Source/Evolve/WarpXEvolve.cpp

```diff
@@ -971,8 +984,18 @@ WarpX::PushParticlesandDepose (int lev, amrex::Real cur_time, DtType a_dt_type,
                 ApplyInverseVolumeScalingToChargeDensity(charge_buf[lev].get(), lev-1);
             }
         }
-    }
+// #else
+//         if (rho_fp[lev].get()) {
+//             // Reflect density over PEC boundaries, if needed.
+//             WarpX::GetInstance().ApplyRhofieldBoundary(lev, rho_fp[lev].get());
+//         }
+//         // Set current density at PEC boundaries, if needed.
+//         WarpX::GetInstance().ApplyJfieldBoundary(
+//             lev, current_fp[lev][0].get(), current_fp[lev][1].get(),
+//             current_fp[lev][2].get()
+//         );
```

Comment:
I left this commented out code here as a reminder that currently the boundary handling for cartesian grids are not matching the RZ handling (done in the `ApplyInverseScalingToChargeDensity` function). The Cartesian grid code had to be moved from here to after the application of the filter to avoid incorrect results. Might this be related to why the filter does not work in RZ (see #1943)?

Reply:
@roelof-groenewald 
Thank you, I think this is a good point. Would you like to add a similar comment directly to #1943, just to keep track of your idea about why the RZ issue might be happening?

Reply:
Sure thing!